apps/api/src/domain/models/media-item/commands/add-generated-transcript-for-media-item/add-generated-transcript-for-media-item.command.spec.ts

```diff
@@ -0,0 +1,208 @@
+                out: 4959.2,
+            },
+            {
+                text: 'hoo',
+                in: 5555.2,
+                out: 5559.34,
+            },
+        ],
+    },
+};
+
+const validFsa: CommandFSA<AddGeneratedTranscriptForMediaItem> = {
+    type: commandType,
+    payload: {
+        aggregateCompositeIdentifier: {
+            type: AggregateType.mediaItem,
+            id: mediaItemId,
+        },
+        source: 'whisper-local',
+        version: '1.5.4',
+        transcript: rawTranscript,
+    },
+};
+
+describe(commandType, () => {
+    let app: INestApplication;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app, databaseProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }).catch((error) => {
+                throw error;
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterAll(async () => {
+        await app.close();
+
+        databaseProvider.close();
+    });
+
+    describe(`when the command is valid`, () => {
+        it(`should succeed with the expected state updates`, async () => {
+            await assertCommandSuccess(commandAssertionDependencies, {
+                systemUserId: dummySystemUserId,
+                buildValidCommandFSA: () => validFsa,
+                seedInitialState: async () => {
+                    await commandAssertionDependencies.testRepositoryProvider
+                        .forResource(ResourceType.mediaItem)
+                        .create(existingMediaItem);
+                },
+                checkStateOnSuccess: async () => {
+                    const updatedMediaItem =
+                        (await commandAssertionDependencies.testRepositoryProvider
+                            .forResource(ResourceType.mediaItem)
+                            .fetchById(mediaItemId)) as MediaItem;
+
+                    const { transcripts } = updatedMediaItem;
+
+                    expect(transcripts).toHaveLength(1);
+
+                    expect(transcripts[0].data).toEqual(rawTranscript);
+
+                    assertEventRecordPersisted(
+                        updatedMediaItem,
+                        'GENERATED_TRANSCRIPT_ADDED_FOR_MEDIA_ITEM',
+                        dummySystemUserId
+                    );
+                },
+            });
+        });
+    });
+
+    describe(`when the command is invalid`, () => {
```

Comment:
Missing cases:
when the MIMEType is not consistent with an audio or video item.